### PR TITLE
🌱 Update versions doc adding Kubernetes 1.26

### PR DESCRIPTION
**Pull request by aniruddha2000**
Signed-off-by: Aniruddha Basak <codewithaniruddha@gmail.com>

part of #7671 

cc @kubernetes-sigs/cluster-api-release-team

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
Looks good
/lgtm


Let's just
/hold
until the other PRs are merged (aka we actually support and test v1.26)

File: docs/book/src/reference/versions.md
Comment (fabriziopandini) [inline docs/book/src/reference/versions.md:78]: If I got it right from the test infra PR, we are adding test coverage for 1.26 also in 1.2, is that right? should we document this as well here?
(same below)

Comment (aniruddha2000) [reply docs/book/src/reference/versions.md:78]: Yes that make sense!

**Comment by sbueringer** [comment]
Let's decide somewhere if we want to commit to supporting v1.26 also in v1.3 and then update the affected PRs accordingly

**Comment by sbueringer** [comment]
> Let's decide if we want to commit to supporting v1.26 also in v1.2 and then update the affected PRs accordingly

Added a comment here: https://github.com/kubernetes-sigs/cluster-api/issues/7671#issuecomment-1369103963

Given that:
/lgtm

(let's merge after the quickstart PR)

**Pull request merged**
